src/test/scala/com/typesafe/scalalogging/LoggerTakingImplicitSpec.scala

```diff
@@ -362,32 +361,32 @@ class LoggerTakingImplicitSpec extends AnyWordSpec with Matchers with Varargs {
     }
 
     "not call the underlying logger's trace method if the trace level is not enabled" in {
-      val f = fixture(_.isTraceEnabled, isEnabled = false)
+      val f = fixture(_.isTraceEnabled, isEnabled = false, stubCanLog = false)
       import f._
       logger.trace(msg, arg1)
       verify(underlying, never).trace(logMsg, arg1)
       logger.trace(msg, arg1, arg2)
       verify(underlying, never).trace(logMsg, forceVarargs(arg1, arg2): _*)
       logger.trace(msg, arg1, arg2, arg3)
       verify(underlying, never).trace(logMsg, arg1, arg2, arg3)
-      verify(canLogCorrelationId, never).logMessage(anyString, any[CorrelationId])
+      verify(canLogCorrelationId, never).logMessage(any[String], any[CorrelationId])
       verify(canLogCorrelationId, never).afterLog(any[CorrelationId])
     }
   }
 
-  def fixture(p: Underlying => Boolean, isEnabled: Boolean) =
+  def fixture(p: Underlying => Boolean, isEnabled: Boolean, stubCanLog: Boolean = true) =
     new {
       implicit val correlationId = CorrelationId("corrId")
-      implicit val canLogCorrelationId = mock(classOf[CanLog[CorrelationId]])
+      implicit val canLogCorrelationId = mock[CanLog[CorrelationId]]
       val msg = "msg"
       val cause = new RuntimeException("cause")
       val arg1 = "arg1"
       val arg2 = Integer.valueOf(1)
       val arg3 = "arg3"
       val logMsg = "corrId - msg"
-      val underlying = mock(classOf[org.slf4j.Logger])
+      val underlying = mock[org.slf4j.Logger]
       when(p(underlying)).thenReturn(isEnabled)
-      when(canLogCorrelationId.logMessage(anyString(), any[CorrelationId])).thenReturn(logMsg)
+      if (stubCanLog) when(canLogCorrelationId.logMessage(any[String], any[CorrelationId])).thenReturn(logMsg)
```

Comment:
Strict mode also means that if you stub something and is not used you'll get a failure as it indicates your code is not doing what the test expected it to do, hence the flag to only stub this for the tests that do a positive verification